test/cpp/api/autograd.cpp

```diff
@@ -123,6 +123,29 @@ TEST(AutogradAPITests, GradUnreachableTest) {
   ASSERT_FALSE(grad_res[1].defined());
 }
 
+TEST(AutogradAPITests, RetainGrad) {
+  auto input = torch::rand({1, 3}, torch::requires_grad());
+  auto h1 = input * 3;
+  auto out = (h1 * h1).sum();
+
+  // It should be possible to call retain_grad() multiple times
+  h1.retain_grad();
+  h1.retain_grad();
+
+  // Gradient should be accumulated
+  out.backward({}, /*keep_graph=*/true);
+  ASSERT_VARIABLE_EQ(h1.data() * 2, h1.grad().data());
```

Comment:
nit: is the `.data()` needed here? `ASSERT_VARIABLE_EQ(h1 * 2, h1.grad());` does not work?

Reply:
My apologies for the oversight - forgot to fix these after the original version of the test passed :(